src/main/java/org/apache/commons/collections4/bloomfilter/SetOperations.java

```diff
@@ -16,147 +16,172 @@
+     * Calculates the Hamming distance between two BitMapProducers.
      *
-     * @param first the first Bloom filter.
-     * @param second the second Bloom filter.
+     * @param first the first BitMapProducer.
+     * @param second the second BitMapProducer.
      * @return the Hamming distance.
      */
-    public static int hammingDistance(final BloomFilter first, final BloomFilter second) {
-        verifyShape(first, second);
-        return first.xorCardinality(second);
+    public static int hammingDistance(final BitMapProducer first, final BitMapProducer second) {
+        return xorCardinality(first, second);
     }
 
     /**
-     * Calculates the Jaccard distance between two Bloom filters.
+     * Calculates the Jaccard distance between two BitMapProducer.
      *
      * <p>Jaccard distance is defined as {@code 1 - Jaccard similarity}</p>
      *
-     * @param first the first Bloom filter.
-     * @param second the second Bloom filter.
+     * @param first the first BitMapProducer.
+     * @param second the second BitMapProducer.
      * @return the Jaccard distance.
      */
-    public static double jaccardDistance(final BloomFilter first, final BloomFilter second) {
+    public static double jaccardDistance(final BitMapProducer first, final BitMapProducer second) {
         return 1.0 - jaccardSimilarity(first, second);
     }
 
     /**
-     * Calculates the Jaccard similarity between two Bloom filters.
+     * Calculates the Jaccard similarity between two BitMapProducer.
      *
      * <p>Also known as Jaccard index, Intersection over Union, and Jaccard similarity coefficient</p>
      *
-     * @param first the first Bloom filter.
-     * @param second the second Bloom filter.
+     * @param first the first BitMapProducer.
+     * @param second the second BitMapProducer.
      * @return the Jaccard similarity.
      */
-    public static double jaccardSimilarity(final BloomFilter first, final BloomFilter second) {
-        verifyShape(first, second);
-        final int orCard = first.orCardinality(second);
-        // if the orCard is zero then the hamming distance will also be zero.
-        return orCard == 0 ? 0 : hammingDistance(first, second) / (double) orCard;
-    }
-
-    /**
-     * Verifies the Bloom filters have the same shape.
-     *
-     * @param first the first filter to check.
-     * @param second the second filter to check.
-     * @throws IllegalArgumentException if the shapes are not the same.
-     */
-    private static void verifyShape(final BloomFilter first, final BloomFilter second) {
-        if (!first.getShape().equals(second.getShape())) {
-            throw new IllegalArgumentException(String.format("Shape %s is not the same as %s",
-                first.getShape(), second.getShape()));
-        }
+    public static double jaccardSimilarity(final BitMapProducer first, final BitMapProducer second) {
+        final int intersection = andCardinality(first, second);
```

Comment:
There is chance for efficiency here. Using my updated method:
```Java
    public static double jaccardSimilarity(final BitMapProducer first, final BitMapProducer second) {
        int[] cardinality = new int[2];
        first.forEachBitMapPair(second, (x, y) -> {
            cardinality[0] += Long.bitCount(x & y);
            cardinality[1] += Long.bitCount(x | y);
            return true;
        });
        final int intersection = cardinality[0];
        return intersection == 0 ? 0 : intersection / (double) cardinality[1];
    }
```
It passes all the current tests in SetOperations.